components/drawer/DrawerContent.js

```diff
@@ -19,8 +20,8 @@ const DrawerContent = (props) => {
       variant="dark" size="medium"
       onPress={async () => {
         Actions.auth();
-        await AsyncStorage.removeItem("token");
-        removePushToken(props.auth.userId);
+        TokenService.clearToken();
+        removeNotificationToken(props.auth.userId);
         if (props.naloxoneAvailability)
           props.setStatus(props.auth.userId, props.auth.token, { "online": false })
```

Comment:
We can't clear the tokens before doing the final setStatus. If props.auth.token is expired, we'll still need the refresh token from async storage